include/pybind11/detail/common.h

```diff
@@ -17,8 +17,60 @@
 // Additional convention: 0xD = dev
 #define PYBIND11_VERSION_HEX 0x020B00D1
 
-#define PYBIND11_NAMESPACE_BEGIN(name) namespace name {
-#define PYBIND11_NAMESPACE_END(name) }
+// You should never need to push and pop manually as these are done in namespace begin and end
```

Comment:
nit In line w/ Ralf's comments, would you be OK with encoding the desire to minimize global suppressions / make things as targeted as possible? e.g.
```cc
// If you find you need to suppress a warning, please try to make the suppression as local as possible
// using these macros. Please also be sure to push/pop with the pybind11 macros.
// Please only use compiler specifics if you need to check specific versions, e.g. Apple Clang vs. vanilla Clang.
```

Reply:
Done!